src/blockserver/server.py

```diff
@@ -105,7 +106,7 @@ def initialize(self, transfer_cls, get_auth_cls, get_cache_cls, database_pool,
         self.streamer = None
         await self._authorize_request()
         if self.request.method == 'POST':
-            self.request.connection.set_max_body_size(options.max_body_size)
+            self.remaining_upload_size = options.max_body_size
```

Comment:
Hm, I think the logic now should be similar. I'd expect that tornado just closes the connection when the set_max_body_size is exceeded; this could has the same behaviour (can't send a response when the request was not fully received, so the HTTPError raised below will just close the connection).